invoices.go

```diff
@@ -172,46 +172,23 @@ func (i *InvoiceCollection) UnmarshalXML(d *xml.Decoder, start xml.StartElement)
 			XMLName xml.Name `xml:"charge_invoice,omitempty"`
 			invoiceFields
 		} `xml:"charge_invoice,omitempty"`
+		CreditInvoices []struct {
+			XMLName xml.Name `xml:"credit_invoice,omitempty"`
+			invoiceFields
+		} `xml:"credit_invoices>credit_invoice,omitempty"`
 	}
 	if err := d.DecodeElement(&v, &start); err != nil {
 		return err
 	}
-	invoice := Invoice{
-		XMLName:               xml.Name{Local: "invoice"},
-		AccountCode:           string(v.ChargeInvoice.AccountCode),
-		Address:               v.ChargeInvoice.Address,
-		OriginalInvoiceNumber: int(v.ChargeInvoice.OriginalInvoiceNumber),
-		UUID:                    v.ChargeInvoice.UUID,
-		State:                   v.ChargeInvoice.State,
-		InvoiceNumberPrefix:     v.ChargeInvoice.InvoiceNumberPrefix,
-		InvoiceNumber:           v.ChargeInvoice.InvoiceNumber,
-		PONumber:                v.ChargeInvoice.PONumber,
-		VATNumber:               v.ChargeInvoice.VATNumber,
-		DiscountInCents:         v.ChargeInvoice.DiscountInCents,
-		SubtotalInCents:         v.ChargeInvoice.SubtotalInCents,
-		TaxInCents:              v.ChargeInvoice.TaxInCents,
-		TotalInCents:            v.ChargeInvoice.TotalInCents,
-		BalanceInCents:          v.ChargeInvoice.BalanceInCents,
-		Currency:                v.ChargeInvoice.Currency,
-		DueOn:                   v.ChargeInvoice.DueOn,
-		CreatedAt:               v.ChargeInvoice.CreatedAt,
-		UpdatedAt:               v.ChargeInvoice.UpdatedAt,
-		AttemptNextCollectionAt: v.ChargeInvoice.AttemptNextCollectionAt,
-		ClosedAt:                v.ChargeInvoice.ClosedAt,
-		Type:                    v.ChargeInvoice.Type,
-		Origin:                  v.ChargeInvoice.Origin,
-		TaxType:                 v.ChargeInvoice.TaxType,
-		TaxRegion:               v.ChargeInvoice.TaxRegion,
-		TaxRate:                 v.ChargeInvoice.TaxRate,
-		NetTerms:                v.ChargeInvoice.NetTerms,
-		CollectionMethod:        v.ChargeInvoice.CollectionMethod,
-		LineItems:               v.ChargeInvoice.LineItems,
-		Transactions:            v.ChargeInvoice.Transactions,
-		CreditPayments:          v.ChargeInvoice.CreditPayments,
+	chargeInvoice := v.ChargeInvoice.ToInvoice()
```

Comment:
I can't comment above this because there is no diff on that line, but we still have the same problem as before (conversion to invoice is happening in multiple places).

On L121 above I think you can change to `*i = v.ToInvoice()`

It's in this function: `func (i *Invoice) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error`

Reply:
done, sorry missed on this previously